pyServer/AzureDiskInspectService.py

```diff
@@ -212,47 +228,116 @@ def ParseUrlArguments(self, urlPath, sasKey):
+        sas_service.create_blob_from_path(self.destinaion_container_name, self.destinaion_blob_name, file_name_full_path)
+
     """
     Upload a local file as a HTTP binary response.
     """
-    def uploadFile(self, http_headers, outputFileName, isPartial, osType):
+    def uploadFile(self, http_headers, outputFileName, isPartial, osType, blobSasUrl):
         # Set the HTTP headers, including extended content from GuestFishWrapper
         self.wfile.write(bytes('HTTP/1.1 200 OK\r\n', 'utf-8'))
-        self.wfile.write(bytes('Content-Type: application/zip\r\n', 'utf-8'))
+
         for header in http_headers:
             header_value = str(http_headers[header])
             if len( header_value ) > 0:
                 self.wfile.write(bytes( '{0}: {1}\r\n'.format( header,header_value ), 'utf-8' ))
                 self.telemetryLogger.info('GuestFishWrapper Header "' + header + '" = "' + header_value +'"')
 
-        statinfo = os.stat(outputFileName)
-        self.wfile.write(bytes('Content-Length: {0}\r\n'.format(
-            str(statinfo.st_size)), 'utf-8'))
-
-        strOutputFileName = os.path.basename(outputFileName) + "-" + osType
-        if isPartial:
-            strOutputFileName = strOutputFileName + "-partial"
-
-        self.wfile.write(bytes(
-            'Content-Disposition: Attachment; filename={0}\r\n'.format(
-                os.path.basename(strOutputFileName)), 'utf-8'))
-        self.wfile.write(bytes('\r\n', 'utf-8'))
-        self.wfile.flush()
-
-        # Write the zip file payload
-        outputFileSize = os.path.getsize(outputFileName)
-        readSize = 0
-        with open(outputFileName, 'rb') as outputFileObj:
-            buf = None
-            while (True):
-                buf = outputFileObj.read(64 * 1024)
-                if (not buf):
-                    break
-                readSize = readSize + len(buf)
-                printProgress(
-                    readSize, outputFileSize, prefix='Progress:', suffix='Complete', barLength=50)
-                self.wfile.write(buf)
-        self.wfile.flush()
+        if blobSasUrl:
+            self.telemetryLogger.info('Uploading result to Blob storage.')
+            self.wfile.write(bytes('Content-Type: text/plain\r\n', 'utf-8'))
+            statinfo = os.stat(outputFileName)
+            self.wfile.write(bytes('Content-Length: 0\r\n', 'utf-8'))
+            self.wfile.write(bytes('\r\n', 'utf-8'))
+            self.wfile.flush()
+
+            local_path=os.path.expanduser("~\libguestfs")
+            full_path_to_file =os.path.join(local_path, outputFileName)
+            file_name = os.path.basename(outputFileName)            
+
+            self.telemetryLogger.info('Uploading: ' + full_path_to_file + ' to Blob...')
+            self.telemetryLogger.info('Filename: ' + file_name)
+
+            self.uploadFileWithBlobSasUrl(file_name_full_path=outputFileName)
+            self.telemetryLogger.info('Uploading to Blob completed.')
```

Comment:
can we time how long it takes to do the upload and log it along with "Uploading to blob completed"?

Reply:
Added time tracking